plugins/auth-backend-module-microsoft-provider/src/strategy.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PassportProfile } from '@backstage/plugin-auth-node';
+import { decodeJwt } from 'jose';
+import fetch from 'node-fetch';
+import { Strategy as MicrosoftStrategy } from 'passport-microsoft';
+
+export class ExtendedMicrosoftStrategy extends MicrosoftStrategy {
+  userProfile(
+    accessToken: string,
+    done: (err?: Error | null, profile?: PassportProfile) => void,
+  ): void {
+    if (this.skipUserProfile(accessToken)) {
+      done(null, undefined);
+      return;
+    }
+
+    super.userProfile(
+      accessToken,
+      (err?: Error | null, profile?: PassportProfile) => {
+        if (!profile || profile.photos) {
+          done(err, profile);
+          return;
+        }
+
+        this.getProfilePhotos(accessToken).then(photos => {
+          profile.photos = photos;
+          done(err, profile);
+        });
+      },
+    );
+  }
+
+  private hasGraphReadScope(accessToken: string): boolean {
+    const { aud, scp } = decodeJwt(accessToken);
+    return (
+      aud === '00000003-0000-0000-c000-000000000000' &&
+      !!scp &&
+      (scp as string)
+        .split(' ')
+        .map(s => s.toLowerCase())
```

Comment:
since it probably doesn't affect us here in practice, i'll defer to a future improvement: should always use `toLocaleLowerCase('en-US')` to ensure that we don't get surprised by e.g. the Turkish locale

Reply:
The original implementation had the same. However, I took care of the adjustment.

btw: There are a couple of other similar places in the codebase.